R/vdiagram-render.R

```diff
@@ -54,6 +54,129 @@ renderVDiagram <- function(data){
 }
 
 createVdiagram <- function(data) {
-  data <- parseVDiagramData(data)
-  vplot(data)
+  styles <- getVDiagramStyle()
+  
+  vdiagramData <- parseVDiagramData(data)
+  
+  vplot <- gsplot() %>%
+    points(NA,NA, ylab=styles$labels$y, xlab=styles$labels$x) %>%
+    callouts(y=c(vdiagramData$minStage, vdiagramData$maxStage), styles$callouts) %>%
+    grid(styles$grid) %>%
+    axis(styles$axis);
+  
+  vplot <- addMeasurementsAndError(vplot, vdiagramData, styles)
+  vplot <- addRatingShifts(vplot, vdiagramData, styles)
+  
+  if (any(!is.na(vdiagramData$obsShift)) && any(!vdiagramData$histFlag)){
+    vplot <- callouts(vplot,x = vdiagramData$obsShift[!vdiagramData$histFlag], y = vdiagramData$obsGage[!vdiagramData$histFlag], 
+                      labels=vdiagramData$obsCallOut[!vdiagramData$histFlag], cex=0.6)
+  }
+  
+  par(mar=c(7, 3, 4, 2))
+  print(vplot) 
+}
+
+addMeasurementsAndError <- function(vplot, vdiagramData, styles) {
+  histFlag <- vdiagramData$histFlag
+  if (any(histFlag)){
+    vplot <- do.call(arrows, append(list(object=vplot, x0=vdiagramData$minShift[histFlag], y0=vdiagramData$obsGage[histFlag], 
```

Comment:
I just had a hard time converting to error_bars and simply punted. I had thought I put a comment in there to eventually do it, must have been overzealous in my house cleaning.